src/UI/PlayerPanel.cpp

```diff
@@ -0,0 +1,370 @@
+{
+    for (auto it = _ui.begin(); it != _ui.end(); ++it)
+    {
+        (*it)->render();
+    }
+
+    // object in hand
+    if (auto item = Game::getInstance()->player()->currentHandSlot())
+    {
+        auto itemUi = item->inventoryDragUi();
+        itemUi->setPosition(position() + Point(360, 60) - itemUi->size() / 2);
+        itemUi->render();
+    }
+}
+
+void PlayerPanel::handle(Event::Event *event)
+{
+    UI::Base::handle(event);
+
+    // object in hand
+    if (auto item = Game::getInstance()->player()->currentHandSlot())
+    {
+        auto itemUi = item->inventoryDragUi();
+        itemUi->handle(event);
+    }
+
+    for (auto it = _ui.rbegin(); it != _ui.rend(); ++it)
+    {
+        if (event->handled()) return;
+        (*it)->handle(event);
+    }
+}
+
+void PlayerPanel::think()
+{
+    UI::Base::think();
+
+    auto game = Game::getInstance();
+
+    for (auto it = _ui.begin(); it != _ui.end(); ++it)
+    {
+        (*it)->think();
+    }
+
+    // object in hand
+    if (auto item = game->player()->currentHandSlot())
+    {
+        auto itemUi = item->inventoryDragUi();
+        itemUi->think();
+    }
+
+}
+
+void PlayerPanel::playWindowOpenSfx()
+{
+    Game::getInstance()->mixer()->playACMSound("sound/sfx/ib1p1xx1.acm");
+}
+
+void PlayerPanel::changeHand()
+{
+    auto player = Game::getInstance()->player();
+    player->setCurrentHand(player->currentHand() == HAND::RIGHT ? HAND::LEFT : HAND::RIGHT);
+    playWindowOpenSfx();
+
+}
+
+void PlayerPanel::openGameMenu()
+{
+    Game::getInstance()->pushState(new State::GameMenu());
+    playWindowOpenSfx();
+}
+
+void PlayerPanel::openInventory()
+{
+    Game::getInstance()->pushState(new State::Inventory());
+    playWindowOpenSfx();
+}
+
+void PlayerPanel::openSkilldex()
+{
+    Game::getInstance()->pushState(new State::Skilldex());
+    playWindowOpenSfx();
+}
+
+void PlayerPanel::openMap()
+{
+    Game::getInstance()->pushState(new State::WorldMap());
+    playWindowOpenSfx();
+}
+
+void PlayerPanel::openCharacterScreen()
+{
+    Game::getInstance()->pushState(new State::PlayerEdit());
+    playWindowOpenSfx();
+}
+
+void PlayerPanel::openPipBoy()
+{
+    Game::getInstance()->pushState(new State::PipBoy());
+    playWindowOpenSfx();
+}
+
+void PlayerPanel::onKeyUp(Event::Keyboard *event)
+{
+    switch (event->keyCode())
+    {
+        case SDLK_a:
+            // @TODO: initiateCombat();
+            break;
+        case SDLK_c:
+            openCharacterScreen();
+            break;
+        case SDLK_i:
+            openInventory();
+            break;
+        case SDLK_p:
+            if (event->controlPressed())
+            {
+                // @TODO: pause game
+            }
+            else
+            {
+                openPipBoy();
+            }
+            break;
+        case SDLK_z:
+            openPipBoy(); // @TODO: go to clock
+            break;
+        case SDLK_ESCAPE:
+        case SDLK_o:
+            openGameMenu();
```

Comment:
I think some key presses that are not directly tied to PlayerPanel can be moved to State::Location. Like GameMenu, SaveGame, LoadGame, Exit,